### PR TITLE
Add bare project

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

Need a place to test modules that aren't a part of the managed workflow

# How

- `expo init` with the bare typescript project
- A lot of tweaking

# Test Plan

- [x] Runs on iOS
- [x] Runs on Android
- [x] Runs on web

### CODE REVIEW

**Comment by ide** [comment]
This is part of https://github.com/expo/expo/issues/4573 -- as follow-up steps it'd be neat to get test-suite and NCL's JS (share the JS instead of copy+paste) running in our bare app, and also add a CI job to build the bare app.

**Comment by EvanBacon** [comment]
In order to run this on Android you need to `export ANDROID_HOME=/Users/evanbacon/Library/Android/sdk` or whatever value `ANDROID_SDK` is. Then you must run `./gradlew :ReactAndroid:installArchives` in `react-native-lab/react-native` this will generate the `/android` folder (~10 minutes). Finally you'll need to run yarn again in the root to create the symlinks.

If you get an error in the Metro server regarding **hasteImpl** or jest you need to run `yarn clear-metro` in `apps/bare-expo` then restart the server with `yarn start`.

**Comment by EvanBacon** [comment]
All of the modules we remove and why:
- [Custom] Remove `expo-task-manager` which doesn't work on Android
  - expo-task-manager
  - unimodules-task-manager-interface
  - expo-background-fetch
  - expo-location
- [Custom] Remove branch
  - expo-branch"
- [Custom] Remove the camera
  - expo-camera
  - expo-face-detector
  - unimodules-face-detector-interface
  - expo-payments-stripe
- [Custom] FBSDK adds a lot of time to the pod install 
  - expo-facebook
  - expo-ads-facebook
- [Custom] To avoid dealing with this error: **'GADInvalidInitializationException', reason: 'The Google Mobile Ads SDK was initialized without an application ID. Google AdMob publishers, follow instructions here: https://googlemobileadssdk.page.link/admob-ios-update-plist to set GADApplicationIdentifier with a valid App ID. Google Ad Manager publishers, follow instructions here: https://googlemobileadssdk.page.link/ad-manager-ios-update-plist'**
  - expo-ads-admob
- [Custom] Remove non-unimodules
  - expo-video-thumbnails
  - expo-module-template

File: apps/bare-expo/metro.config.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const path = require('path');
+const { createMetroConfiguration } = require('expo-yarn-workspaces');
+
+const blacklist = require('metro-config/src/defaults/blacklist');
+const config = createMetroConfiguration(__dirname);
+
+const blacklistRE = () =>
```
Comment (ide) [inline apps/bare-expo/metro.config.js]: Instead of redefining the blacklist here, it seems like we should update the config that e-y-w creates. That said, if we already define the blacklist in the `resolver` configuration it seems like it should therefore filter out the blacklisted files from ever getting transformed too.

File: apps/bare-expo/metro.config.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  blacklist([
+    // [Custom] Prevent the haste collision in `ios/` versioned code.
+    // Otherwise you will get an error from Metro bundler.
+    /.*\/android\/ReactAndroid\/.*/,
+    /.*\/versioned-react-native\/.*/,
+  ]);
+
+module.exports = {
+  ...config,
+  watchFolders: [
```
Comment (ide) [inline apps/bare-expo/metro.config.js]: e-y-w watches the workspace root -- does that cause issues for the bare project?

File: apps/bare-expo/metro.config.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+const path = require('path');
+const { createMetroConfiguration } = require('expo-yarn-workspaces');
+
+const blacklist = require('metro-config/src/defaults/blacklist');
+const config = createMetroConfiguration(__dirname);
```
Comment (ide) [inline apps/bare-expo/metro.config.js]: We use this config for all of the other apps -- is it that we load application code from `apps/ncl` that necessitates a custom config?

File: apps/bare-expo/android/keystores/debug.keystore.properties [hunk 4]
```diff
@@ -0,0 +1,4 @@
+key.store=debug.keystore
```
Comment (ide) [inline apps/bare-expo/android/keystores/debug.keystore.properties]: For Google Maps and other Google APIs, we need to use a shared keystore during development. We already have a debug keystore that we use for the Expo client -- whether it's through this properties file or the Gradle file, make the debug build use our shared keystore.

**Review by ide — CHANGES_REQUESTED** [review]
Rename the iOS bundle identifier to `dev.expo.Payments` and the Android package name to `com.expo.payments` so that the IAPs are set up.

**Comment by ide** [comment]
The Android build folder also needs to be added to a new .gitignore.

File: apps/bare-expo/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
+{
```
[the remark this answers is not in the document]
Comment (EvanBacon) [reply apps/bare-expo/tsconfig.json]: tryna keep it as close to bare RN as possible though

File: apps/bare-expo/android/keystores/debug.keystore.properties (hunk 4 above)
Comment (EvanBacon) [reply apps/bare-expo/android/keystores/debug.keystore.properties]: ELI5
